examples/vue-dapp/src/components/Main.vue

```diff
@@ -0,0 +1,172 @@
+<template>
+  <div>
+    <div v-if="isLoading">
+      <Spinner />
+      <span>
+        Loading... (Please sign messages using your connected wallet)
+      </span>
+    </div>
+    <div v-else-if="did">
+      <p>Hello user!</p>
+      <p>
+        Your decentralised identifier: <strong>{{ did }}</strong>
+      </p>
+      <div v-if="roles.length > 0" class="rolesContainer">
+        <p>This are your validated roles:</p>
+        <p v-for="role in roles" v-bind:key="role.name">
+          <strong>{{ role.name }}</strong> at {{ role.namespace }}
+        </p>
+      </div>
+      <div v-else>
+        You do not have any issued role at the moment, please login into switchboard and search for
+        apps, orgs to enrol.
+      </div>
+      <div class="logoutContainer">
+        <button @click="logout" class="button">
+          <span>Logout</span>
+        </button>
+      </div>
+    </div>
+    <div v-else class="container">
+      <button @click="loginWithWalletConnect" class="button">
+        <img class="walletconnect" src="../assets/wallet-connect-icon.svg" />
+        <span>Login with Wallect Connect</span>
+      </button>
+      <button @click="loginWithMetamask" class="button">
+        <img class="metamask" src="../assets/metamask-logo.svg" />
+        <span>Login with Metamask</span>
+      </button>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+import Spinner from "./Spinner.vue";
+import axios from "axios";
+
+type Role = {
+  name: string;
+  namespace: string;
+};
+
+type User = {
```

Comment:
Seems as though the `User` type isn't being used. If so, can it be removed?

Reply:
Fixed 😄